args_test.go

```diff
@@ -151,3 +151,23 @@ func TestArgsUsage(t *testing.T) {
 		})
 	}
 }
+
+func TestSingleOptionalArg(t *testing.T) {
+	cmd := buildMinimalTestCommand()
+	var s1 string
+	arg := &StringArg{
+		Min:         0,
+		Max:         1,
+		Destination: &s1,
+	}
+	cmd.Arguments = []Argument{
+		arg,
+	}
+
+	require.NoError(t, cmd.Run(context.Background(), []string{"foo"}))
+	require.Equal(t, "", s1)
+
+	arg.Value = "bar"
+	require.NoError(t, cmd.Run(context.Background(), []string{"foo"}))
```

Comment:
could you also add a test for 

```
[]string{"foo", "zbar"}
```

to make sure that value is picked up from args

Reply:
all set!